### ISSUE TITLE
Cryptic Error when Uploading Projects without Themes

### ISSUE THREAD

**Issue by shreeramsenthi** [NCC-CNC/wheretowork#281]
Uploading project data including a YAML file without themes leads to a cryptic error ("Error : x is not a list") with no details on fixing the issue. It was easy to accidentally produce project data like this with previous versions of the PRZTools prototype. Although error checking has been added to PRZTools to catch this issue, it would be best to provide more information here as well.

This issue might result from trying to access members of the "themes" without first checking that it is a list. One option would be catch missing themes as part of the [configuration file validation](https://github.com/NCC-CNC/wheretowork/blob/100895f32767d31f1d8d59c5d4656a8485debcfc/R/utils_file_validation.R#L13) on upload.

Attached are a zip of the inputs and the output log file with a minimal reproducible example generated using an older version of PRZTools.

[MRE Inputs.zip](https://github.com/NCC-CNC/wheretowork/files/10491713/MRE.Inputs.zip)
[WTW_CONFIG_log.zip](https://github.com/NCC-CNC/wheretowork/files/10491720/WTW_CONFIG_log.zip)

**Comment by DanWismer** [comment]
Hi @shreeramsenthi, 

I was able to pin-point where the error occurs. The `fct_read_project.R` on line 438 calls the `SolutionSetting` method `ss$update_current_held()`. This triggers the  `get_theme_data()` method in `class_SolutionSettings.R` on line 392; which for themes, a `Matrix::sparseMatrix()` is suppose to be returned. However, if there are no themes (i.e. an empty list), an error occurs on line 401 on `class_SolutionSettings.R`.

Long story short, every project must have at least 1 theme. I will need to propagate that error message up the pipeline so that is more obvious for the user. Thanks for flagging this!

Also, looks like you are working on the Pro data-prep plug-in? Feel free to ask as many questions as you need.

**Comment by shreeramsenthi** [comment]
Thanks for the quick response Dan! I can see why people hadn't run into it before, because it's pretty nonsensical to not include a theme in a project and the current file validation already raises an error if there is no "theme" element in the YAML. This was mostly an edge case with the plug-in.

And thanks! I'll probably take you up on that at some point! I'm still pretty new to working with prioritizr / WTW
